src/Propel/Common/Config/XmlToArrayConverter.php

```diff
@@ -80,8 +78,6 @@ public static function convert($xmlToParse)
      *
      * @author Christophe VG (based on code form php.net manual comment)
      *
-     * @psalm-return array<scalar, mixed>
-     *
      * @param \SimpleXMLElement $xml SimpleXML object.
      *
      * @return array Array representation of SimpleXML object.
```

Comment:
Can't you just use

    @return array<int|string,mixed>

instead etc?
Then we can avoid a non functional cast (as this is clear for keys given anyway) - we would only do it for phpstan.

Reply:
Or just remove the call to `getConvertedXmlValue()`, and replace it with something like
```php
$k = (ctype_digit($av)) ? (int)$av : $av;
```
If you set an id to `trUe`, you don't want it to become `true`.

Nice find though! I remember not really understanding the psalm error.

Reply:
done